cpp/src/gandiva/gdv_function_stubs.cc

```diff
@@ -794,6 +796,21 @@ const char* gdv_fn_initcap_utf8(int64_t context, const char* data, int32_t data_
   *out_len = out_idx;
   return out;
 }
+
+GANDIVA_EXPORT
+int32_t gdv_fn_instr_utf8(int64_t context, const char* string, int32_t string_len,
+                         const char* substring, int32_t substring_len) {
+  if (substring_len == 0) {
+    return 1;
+  }
+
+  for(int i = 0; i < string_len; i++) {
```

Comment:
The condition (i < string_len) is incorrect. Note that you are comparing the next N bytes when the first byte matches. If there is a match at the last byte, you will access beyond the end here and cause issues

This should be changed to (i < string_len - substring_len)

Reply:
Fixed